actions/inference-mock/app.py

```diff
@@ -0,0 +1,81 @@
+# Standard
+from dataclasses import dataclass
+import logging
+import pprint
+
+# Third Party
+from completions.completion import create_chat_completion
+from flask import Flask, request  # type: ignore[import-not-found]
+from matching.matching import Matcher
+from werkzeug import exceptions  # type: ignore[import-not-found]
+import click  # type: ignore[import-not-found]
+import yaml
+
+# Globals
+app = Flask(__name__)
+strategies: Matcher  # a read only list of matching strategies
+
+
+# Routes
+@app.route("/v1/completions", methods=["POST"])
+def completions():
+    data = request.get_json()
+    if not data or "prompt" not in data:
+        raise exceptions.BadRequest("prompt is empty or None")
+
+    prompt_debug_str = data["prompt"][:90] + "..."
+    app.logger.debug(
+        f"{request.method} {request.url} {data['model']} {prompt_debug_str}"
+    )
+
+    prompt = data.get("prompt")
+    chat_response = strategies.find_match(
+        prompt
+    )  # handle prompt and generate correct response
+
+    response = create_chat_completion(chat_response, model=data.get("model"))
+    app.logger.debug(f"response: {pprint.pformat(response, compact=True)}")
+    return response
+
+
+# config
+@dataclass
+class Config:
+    matches: list[dict]
+    port: int = 11434
+    debug: bool = False
+
+
+@click.command()
+@click.option(
+    "--config",
+    type=click.Path(),
+    required=True,
+    help="path to a YAML config file containing detailed configuration and model response options",
+)
+def start_server(config):
+    # get config
+    with open(config, "r", encoding="utf-8") as file:
+        yaml_data = yaml.safe_load(file)
+    if not isinstance(yaml_data, dict):
+        raise ValueError("config file format is invalid")
```

Comment:
We can help the user by naming the invalid configuration and giving a clue about what specifically was invalid about the file's contents:
```suggestion
        raise ValueError(f"config file {config} must be a set of key-value pairs")
```